src/routes/trading.routes.ts

```diff
@@ -71,6 +106,64 @@ export const postTrade = async (req: Request, res: Response): Promise<void> => {
   res.end();
 };
 
+async function checkForDuplicate(trade: any): Promise<boolean> {
+  const db = DatabaseService.getDatabaseInstance();
+  const { stub, symbol, chart, TCycles, TBuys } = trade;
+  let existingEntry;
+  if (TCycles && TBuys && chart) {
+    deleteOldRecords(stub, symbol, chart, TCycles);
+    try {
+      existingEntry = await db.read(
+        `${stub}/orders/${symbol}/${chart}/${TCycles}/${TBuys}`
```

Comment:
This could be made more generic by just accepting a `uniqueCode` string in the payload